src/components/formSections/components/TermsListField.js

```diff
@@ -172,6 +172,11 @@ export default class TermsListField extends React.Component {
     if (term.type === TERM_TYPE_NUMBER) {
       FieldComponent = TextField;
       fieldProps.type = 'number';
+      const min = 0;
+      const max = Number.MAX_SAFE_INTEGER;
+      if (currentValue.value < min || currentValue.value > max) {
+        errors[term.value] = <FormattedMessage id="ui-licenses.errors.termValueNotInRange" values={{ min, max }} />;
+      }
```

Comment:
@CalamityC Is there a reason you are validating the Number range here in the `renderTermValue` code black rather than in the `isInvalid` function [here](https://github.com/folio-org/ui-licenses/blob/master/src/components/formSections/components/TermsListField.js#L131). Ideally we would like to pull all the errors in the `TermsListField` from a single object -> [errors state variable](https://github.com/folio-org/ui-licenses/blob/master/src/components/formSections/components/TermsListField.js#L41) rather than setting it inline. That `errors` object is being [set](https://github.com/folio-org/ui-licenses/blob/master/src/components/formSections/components/TermsListField.js#L147) inside the `isInvalid` validate function that gets called [here](https://github.com/folio-org/ui-licenses/blob/master/src/components/formSections/FormTerms.js#L71). 

So, I believe the `currentValue` you are using in the above code block should be the same as `value` [here](https://github.com/folio-org/ui-licenses/blob/master/src/components/formSections/components/TermsListField.js#L136) and you could set the `errors[term.value]` accordingly there.